### PR TITLE
Add Swift 4 support

### PR DESCRIPTION
**Pull request by irace**
I can understand not wanting to merge this into `master` until Xcode 9 is out of beta – you may want to make a `swift-4` branch that can become the base branch for this PR?

### CODE REVIEW

File: Sources/Cache.swift [hunk 1]
```diff
@@ -66,7 +66,7 @@ public final class Cache: Caching {
         #endif
     }
 
-    private static func defaultCostLimit() -> Int {
+    public static func defaultCostLimit() -> Int {
```
Comment (kean) [inline Sources/Cache.swift]: This things used to be private.

File: Sources/Cache.swift [hunk 2]
```diff
@@ -117,7 +117,7 @@ public final class Cache: Caching {
     }
 
     /// Removes all cached images.
-    public dynamic func removeAll() {
+    @objc public dynamic func removeAll() {
```
Comment (kean) [inline Sources/Cache.swift]: I haven't yet went through all of Swift 4 changes. Not sure, why is this change necessary?

File: Sources/Cache.swift (hunk 1 above)
Comment (irace) [inline Sources/Cache.swift]: As of Swift 4, they can’t be private because they’re used as default arguments in public methods (I believe constructors in this specific case).

File: Sources/Cache.swift (hunk 2 above)
Comment (irace) [inline Sources/Cache.swift]: https://github.com/apple/swift-evolution/blob/master/proposals/0160-objc-inference.md

**Comment by kean** [comment]
Thanks!

**Comment by kean** [comment]
FYI going to release a new version with minimum changes soon which supports both Xcode 8 & 9 and both Swift 3.2 & 4.
